### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by user2589**
automatic version numbering and prefix lengths, small refactorings etc

### CODE REVIEW

File: oscar.py [hunk 1]
```diff
@@ -1537,132 +1548,116 @@ class Time_commit_info(Clickhouse_DB):
...
+        | sha1     | FixedString(20) |
         | time     | UInt32          |
-        | author   | String          |
+        | blob     | FixedString(20) |
         | language | String          |
+        | repo     | String          |
+        | author   | String          |
         | deps     | String          |
-    '''
-    columns = ['blob', 'commit', 'project', 'time', 'author', 'language', 'deps']
```
Comment (px1624) [inline oscar.py:1603]: This should be the correct table schema.

File: oscar.py [hunk 2]
```diff
@@ -1537,132 +1548,116 @@ class Time_commit_info(Clickhouse_DB):
...
+                'lower(hex(sha1))', self.tb_name, start, end):
             yield row[0]
-        
+
+
 class Time_project_info(Clickhouse_DB):
-    ''' Time_project_info class is initialized with table name and database host name
-        The default table name for projects is projects_all, and the default database name is localhost
-        This class contains methods to query for project data
-        The 'b2cPtaPkgR_all' table descrption is the following:
```
Comment (px1624) [inline oscar.py:1593]: The table name 'b2cPtaPkgR_all' is the new table, 'projects_all' table has been removed.

File: oscar.py [hunk 3]
```diff
@@ -1537,132 +1548,116 @@ class Time_commit_info(Clickhouse_DB):
...
-        for row in rows_iter:
-            yield row
+        return self.query_iter(query_str)
 
     def __wrap_cols(self, cols):
-        ''' wraps cols to select before querying
-        '''
+        """ wraps cols to select before querying """
         for i in range(len(cols)):
-            if cols[i] == 'commit' or cols[i] == 'blob':
```
Comment (px1624) [inline oscar.py:1666]: the sha1 column no longer exits in the new table, the commit column is the replacement.

File: oscar.py [hunk 4]
```diff
@@ -1537,132 +1548,116 @@ class Time_commit_info(Clickhouse_DB):
...
         | language | String          |
+        | repo     | String          |
+        | author   | String          |
         | deps     | String          |
-    '''
-    columns = ['blob', 'commit', 'project', 'time', 'author', 'language', 'deps']
+    """
+    columns = ['sha1', 'time', 'blob', 'language', 'repo', 'author', 'deps']
 
-    def __init__(self, tb_name='b2cPtaPkgR_all', db_host='localhost'):
```
Comment (px1624) [inline oscar.py:1605]: This should be the correct default table name.

File: oscar.py [hunk 5]
```diff
@@ -1537,132 +1548,116 @@ class Time_commit_info(Clickhouse_DB):
...
-        rows_iter = self.query_iter(query_str)
-        for row in rows_iter:
-            yield row
+        query_str = 'SELECT {} FROM {} WHERE repo=\'{}\' ORDER BY time'\
+                    .format(', '.join(cols), self.tb_name, repo)
+        return self.query_iter(query_str)
 
     def author_timeline(self, cols, author):
-        ''' return a generator for all rows given an author (ordered by time)
-        >>> rows = p.author_timeline(['time', 'project'], 'Andrew Gacek <andrew.gacek@gmail.com>')
```
Comment (px1624) [inline oscar.py:1646]: the 'repo' column has been renamed to 'project'

File: oscar.py (hunk 4 above)
Comment (user2589) [reply oscar.py:1605]: yes, there was a merge conflict I had to resolve manually and I overlooked these changes. I restored the last schema, can you please have a look again?

File: oscar.py [hunk 6]
```diff
@@ -1617,51 +1619,46 @@ def get_values_iter(self, cols, start, end):
...
         ...     print(row)
         ...
         (1568571909, 'mrtrevanderson_CECS_424')
         (1568571909, 'mrtrevanderson_CECS_424')
         (1568571909, 'mrtrevanderson_CECS_424')
         ...
-        '''
+        """
         cols = self.__wrap_cols(cols)
-        query_str = 'SELECT {} FROM {} WHERE project=\'{}\' ORDER BY time'\
```
Comment (px1624) [inline oscar.py:1638]: I think this line should be kept as well, everything else should be fine.

**Pull request merged**